test/unity/toolkits/drawing_classifier/test_dc_serialization.cxx

```diff
@@ -184,47 +185,44 @@ BOOST_AUTO_TEST_CASE(test_load_version) {
   TS_ASSERT_EQUALS(dc.get_version(), dc.DRAWING_CLASSIFIER_VERSION);
 }
 
-
 BOOST_AUTO_TEST_CASE(test_save_load) {
-  auto remove_if_exist =
-      [](const char* fname) {
-        std::FILE* fp = nullptr;
-        if ((fp = std::fopen(fname, "r")) != NULL) {
-          if (std::fclose(fp) != 0) {
-            std::stringstream ss;
-            ss << "cannot close file: " << fname << std::endl;
-            throw(turi::error::io_error(ss.str()));
-          }
-
-          if (std::remove(fname) != 0) {
-            std::stringstream ss;
-            ss << "cannot remove file: " << fname << std::endl;
-            throw(turi::error::io_error(ss.str()));
-          }
-        }
-      };
+  auto remove_if_exist = [](const char* fname) {
+    std::FILE* fp = nullptr;
+    if ((fp = std::fopen(fname, "r")) != NULL) {
+      if (std::fclose(fp) != 0) {
+        std::stringstream ss;
+        ss << "cannot close file: " << fname << std::endl;
+        throw(turi::error::io_error(ss.str()));
+      }
+
+      if (std::remove(fname) != 0) {
+        std::stringstream ss;
+        ss << "cannot remove file: " << fname << std::endl;
+        throw(turi::error::io_error(ss.str()));
+      }
+    }
+  };
 
   /**
-   * This is a hack because nn_spec_ is a private member. Thus, I can't
-   * compare it directly after loading from file (load_version call).
-   *
-   * The solution is to call `load_version` first and write serialized
-   * `nn_spec_` to disk. In this way, I can indirectly verify the `nn_spec_`
-   * successively loads all params from the source model spec file.
+   * The test calls `load_version` first and write serialized
+   * `nn_spec_` to disk. In this way, the `nn_spec_` can be verified if
+   * it successively loads all params from the source model spec file.
    */
-  auto load_save_compare = [&remove_if_exist](drawing_classifier_mock& dc,
-                              drawing_classifier_mock& dc_other) {
+  auto load_save_compare = [&remove_if_exist](
+                               drawing_classifier_mock& dc,
+                               drawing_classifier_mock& dc_other) {
     // random init; avoid segfault
-    dc.my_init_model_spec();
-    dc_other.my_init_model_spec();
+    dc.my_model_init_and_copy();
+    dc_other.my_model_init_and_copy();
```

Comment:
What is this doing? It's not clear to me

Reply:
init model and get a copy of nn_spec.